src/string_immutable.zig

```diff
@@ -39,38 +39,19 @@ pub inline fn w(comptime str: []const u8) [:0]const u16 {
     return Static.literal;
 }
 
-pub fn toUTF16Literal(comptime str: []const u8) []const u16 {
-    return comptime literal(u16, str);
-}
-
 pub inline fn literal(comptime T: type, comptime str: string) []const T {
```

Comment:
i was going to comment about the inComptime thing but it seems that is not needed anymore

i wonder if we should make this function return a sentinel slice since that is what string literals are, or more specifically, pointers to an array terminated by 0

Reply:
I have found the error helpful.

Reply:
the error will still throw but the check was doing redundant work the compiler already does now